Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py

```diff
@@ -133,12 +133,13 @@ class IAMUserProfile:
 
     def __init__(self, user_profile, user_profile_delta=None):
         self._user_profile = safe_load_json(user_profile)
+        self.old_user_data = self._user_profile.get('olduserdata', {})
         self._user_profile_delta = safe_load_json(user_profile_delta) if user_profile_delta else {}
         self._vendor_action_results = []
 
     def get_attribute(self, item, use_old_user_data=False):
-        if use_old_user_data and self._user_profile.get('olduserdata', {}).get(item):
-            return self._user_profile.get('olduserdata', {}).get(item)
+        if use_old_user_data and self.old_user_data.get(item):
+            return self.old_user_data.get(item)
```

Comment:
I think its better to save the old user data as dict instead of another profile (more confusing in my opinion to hold 2 user profiles when we're talking about same user. But if you want us to do so ill change